plugins/uefi/fu-plugin-uefi.c

```diff
@@ -622,16 +610,52 @@ fu_plugin_coldplug (FuPlugin *plugin, GError **error)
 	if (entries == NULL)
 		return FALSE;
 
+	/* if secure boot is enabled ensure we have a signed fwupd.efi */
+	bootloader = fu_uefi_get_built_app_path (&error_bootloader);
+	if (bootloader == NULL)
+		g_warning ("%s", error_bootloader->message);
+
+	/* ensure the ESP is detected */
+	if (!fu_plugin_uefi_ensure_esp_path (plugin, &error_esp))
+		g_warning ("%s", error_esp->message);
+
 	/* add each device */
 	for (guint i = 0; i < entries->len; i++) {
 		const gchar *path = g_ptr_array_index (entries, i);
 		g_autoptr(FuUefiDevice) dev = fu_uefi_device_new_from_entry (path);
 		if (!fu_plugin_uefi_coldplug_device (plugin, dev, error))
 			return FALSE;
-		fu_device_set_metadata (FU_DEVICE (dev), "EspPath", data->esp_path);
+		if (error_esp != NULL) {
+			fu_device_set_update_error (FU_DEVICE (dev), error_esp->message);
+		} else if (error_bootloader != NULL) {
+			fu_device_set_update_error (FU_DEVICE (dev), error_bootloader->message);
+		} else {
+			fu_device_set_metadata (FU_DEVICE (dev), "EspPath", data->esp_path);
+			fu_device_add_flag (FU_DEVICE (dev), FWUPD_DEVICE_FLAG_UPDATABLE);
+		}
 		fu_plugin_device_add (plugin, FU_DEVICE (dev));
 	}
 
+	/* no devices are updatable */
+	if (error_esp != NULL || error_bootloader != NULL)
+		return TRUE;
+
+	/* delete any existing .cap files to avoid the small ESP partition
+	 * from running out of space when we've done lots of firmware updates
+	 * -- also if the distro has changed the ESP may be different anyway */
+	if (fu_uefi_vars_exists (FU_UEFI_VARS_GUID_EFI_GLOBAL, "BootNext")) {
+		g_debug ("detected BootNext, not cleaning up");
+	} else {
+		if (!fu_plugin_uefi_delete_old_capsules (plugin, error))
+			return FALSE;
+		if (!fu_uefi_vars_delete_with_glob (FU_UEFI_VARS_GUID_FWUPDATE, "fwupd-*", error))
+			return FALSE;
+	}
+
+	/* save in report metadata */
+	g_debug ("ESP mountpoint set as %s", data->esp_path);
+	fu_plugin_add_report_metadata (plugin, "ESPMountPoint", data->esp_path);
```

Comment:
As I can tell this is now duplicated with the proxy device registration so this can probably go.

Reply:
Are you sure? I thought plugins didn't handle their own registrations anymore?

Reply:
Oh right, I forgot about that change.  Then yeah this PR looks fine.